firebase-database/src/main/java/com/google/firebase/database/core/utilities/ParsedUrl.java

```diff
@@ -32,4 +32,11 @@ public boolean equals(Object o) {
     if (!repoInfo.equals(parsedUrl.repoInfo)) return false;
     return path.equals(parsedUrl.path);
   }
+
+  @Override
+  public int hashCode() {
+    int result = repoInfo.hashCode();
+    result = 31 * result + path.hashCode();
+    return result;
```

Comment:
```suggestion
    return 31 * repoInfo.hashCode() + path.hashCode();
```

Reply:
It's been too many years since I've been in school to remember the name of the notation but writing
AX^N + BX^(N-1) + CX^(N-2)...YX^1 + Z as
((((A)X + B)X + C)X + D)... 
is a much very common factorization for hashes and checksums because it forms a pattern that works incrementally as the input increase. Keeping the pattern.